src/EditorFeatures/CSharpTest/CodeLens/CSharpCodeLensTests.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeLens;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeLens
+{
+    public class CSharpCodeLensTests
+    {
+        private CodeLensReferenceService _referenceService;
+
+        private async Task<TestWorkspace> SetupWorkspaceAsync(string content)
+        {
+            var workspace = await TestWorkspace.CreateCSharpAsync(content);
+            _referenceService = new CodeLensReferenceService();
+            return workspace;
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeLens)]
+        public async Task TestCount()
+        {
+            using (var workspace = await SetupWorkspaceAsync(@"
+class A
+{
+    void B()
+    {
+        C();
+    }
+
+    void C()
+    {
+        D();
+    }
+
+    void D()
+    {
+        C();
+    }
+}"))
+            {
+                var solution = workspace.CurrentSolution;
```

Comment:
I don't like these tests.  They violate the way we generally do testing in the IDE layer.  Instead, i would expect to see something lke:

```c#
+class A
 +{
       {|references:0|}
 +    void B()
 +    {
 +        C();
 +    }
 +
       {|references:1|}
 +    void C()
 +    {
 +        D();
 +    }
 +
 +    void D()
 +    {
 +        C();
 +    }
 +}"))
```

i.e. we'd use our markup format to specify the types of results we expect.  Then we'd hav ea single test driver that verified things.  You can see our FindReferences tests for examples of what i mean.